pycbc/inference/models/gaussian_noise.py

```diff
@@ -883,6 +923,10 @@ def _loglr(self):
                 return self._nowaveform_loglr()
             else:
                 raise e
+
+        if save_waveforms:
+            self.wfs = {}
+
```

Comment:
Rather than add this, I think it'd be better to move [this function](https://github.com/gwastro/pycbc/blob/master/pycbc/inference/models/gated_gaussian_noise.py#L197-L223) from `BaseGatedGaussianNoise` over to `BaseGaussianNoise`, then modify the `loglr` function currently in `GaussianNoise` to call `get_waveforms` rather than the waveform generator directly. This would also be a useful step toward having a script that plots the maxL waveforms. Also could be useful for my comment above: then the `multi_loglikelihood` could just modify what `get_waveforms` does (maybe through an attribute), then call the loglr function.

Reply:
@cdcapano The function in gated gaussian needs to remain as it all the things in the inner loop aren't appropriate for the regular model. I think I need to leave the function in gated gaussian there, but possibly inherit from the one in gaussian noise (added).